src/DataService/ByFields.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Lullabot\Mpx\DataService;
+
+/**
+ * A collection of fields to filter a request by.
+ *
+ * Results from a ByField query are paged, and MPX enforces paging if none is
+ * specified. Since sorting may be inconsistent across pages, this class will
+ * automatically sort by 'id' if no sort is specified.
+ *
+ * @see https://docs.theplatform.com/help/wsf-selecting-objects-by-using-a-byfield-query-parameter
+ */
+class ByFields
+{
+    /**
+     * The array of fields and their value to filter by.
+     *
+     * @var array
+     */
+    protected $fields = [];
+
+    /**
+     * The sort to apply to this filter.
+     *
+     * @var Sort
```

Comment:
I guess you're using the class name because it belongs to the same namespace we are in. But that this have real implications apart from readibility? I mean, does PhpStorm understand that this variable is \Lullabot\Mpx\DataService\Sort without the use clause?

Reply:
Yes, it does! I don't know when PHPStorm added support for that, but it now works fine as long as the class is imported with a use statement. It surprised me too! Apparently the Symfony standards want you to shorten classes in docs, while Drupal wants the full path.

<img width="778" alt="preferences 2018-03-29 06-58-47" src="https://user-images.githubusercontent.com/255023/38085492-ba7700ba-331e-11e8-8944-1b17e811b0e8.png">

Reply:
> it now works fine as long as the class is imported with a use statement

That's what I mean,  you're not using the use statement!

Reply:
Ah, in that case it works because Sort and ByFields are in the same namespace.